apps/rush-lib/src/logic/PackageJsonUpdater.ts

```diff
@@ -306,67 +309,98 @@ export class PackageJsonUpdater {
 
     await installManager.ensureLocalPackageManager();
     let selectedVersion: string | undefined;
+    const rushConfigProjects: RushConfigurationProject[] = this._rushConfiguration.projects;
 
     if (initialSpec && initialSpec !== 'latest') {
       console.log(colors.gray('Finding versions that satisfy the selector: ') + initialSpec);
       console.log();
-      console.log(`Querying registry for all versions of "${packageName}"...`);
 
-      let commandArgs: string[];
-      if (this._rushConfiguration.packageManager === 'yarn') {
-        commandArgs = ['info', packageName, 'versions', '--json'];
-      } else {
-        commandArgs = ['view', packageName, 'versions', '--json'];
+      // determine if the package is a project in the local repository and if the version exists
+      for (const rushConfigProject of rushConfigProjects) {
+        const rushConfigProjectVersion: string = rushConfigProject.packageJson.version;
+        if(rushConfigProject.packageName === packageName && !projects[0].cyclicDependencyProjects.has(packageName)) {
+          if(semver.satisfies(rushConfigProjectVersion, initialSpec)) {
+            selectedVersion = initialSpec;
+            break;
+          } else {
+            throw new Error(`Unable to find a version of "${packageName}" that satisfies`
```

Comment:
Let's be a little more elaborate with this error:

```suggestion
            throw new Error(`The dependency being added ("${packageName}") is a project in the local Rush repository, but the version specifier provided (${initialSpec}) does not match the local project version (${rushConfigProjectVersion}). Correct the version specifier, omit a version specifier, or include "${packageName}" as a cyclicDependencyProject in rush.json for ${<project-being-added-to>} if it is intended for "${packageName}" to come from an external feed and not from the local Rush repository.`);
```